src/types/update.rs

```diff
@@ -28,6 +29,27 @@ pub struct Update {
     pub kind: UpdateKind,
 }
 
+impl Update {
+    /// Tries to parse `value` into `Update`, logging an error on failure.
+    ///
+    /// It is used to implement update listeners.
+    pub fn try_parse(value: &Value) -> Result<Self, serde_json::Error> {
+        match serde_json::from_str(&value.to_string()) {
```

Comment:
Can you add a comment explaining why [`from_value`](https://docs.rs/serde_json/1.0.64/serde_json/fn.from_value.html) isn't used here?

Reply:
Oh, actually it can be used here.